src/coreclr/vm/i386/stublinkerx86.cpp

```diff
@@ -4541,29 +4541,43 @@ VOID StubLinkerCPU::EmitArrayOpStub(const ArrayOpScript* pArrayOpScript)
                         total += cur->startoffset - elemOfs;
 
                         SSIZE_T cnt = (SSIZE_T) pArrayOpScript->m_gcDesc->GetNumSeries();
-                        // special array encoding
-                        _ASSERTE(cnt < 0);
 
-                        for (SSIZE_T __i = 0; __i > cnt; __i--)
+                        _ASSERTE(cnt < 0 || cnt == 1);
+                        if (cnt == 1)
                         {
-                            HALF_SIZE_T skip =  cur->val_serie[__i].skip;
-                            HALF_SIZE_T nptrs = cur->val_serie[__i].nptrs;
-                            total += nptrs*sizeof (DWORD*);
-                            do
+                            // all pointers
+                            for (size_t i = 0; i < size; i += sizeof(DWORD*))
```

Comment:
```suggestion
                            for (size_t i = 0; i < size; i += sizeof(Object*))
```
More self-explanatory

Reply:
this file has no consistency on the matter of pointer sizes - some code above uses `sizeof(void*)`, some code closer to here uses `DWORD*`, so I chose that, although `DWORD*` is indeed a bit unusual.

The formatting style ( i.e. `total += nptrs*sizeof (DWORD*);` ) looks like this code might be pretty old code.